### PR TITLE
LPD-40990 Clear object action thread local when adding system object entries, the same way it's done for custom object entries

### PR DESCRIPTION
**Pull request by devpedrotavares**
This is a follow up on: https://github.com/brianchandotcom/liferay-portal/pull/155958. 

I have applied the feedback on the tests. And also added a call to `clearObjectEntryIdsMap` on `onAfterRemove` to fix the problem for the deletion case.

Related ticket: [LPD-40990](https://liferay.atlassian.net/browse/LPD-40990)

[LPD-40990]: https://liferay.atlassian.net/browse/LPD-40990?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java [hunk 1]
```diff
@@ -2159,6 +1942,14 @@ public void testDeleteObjectAction() throws Exception {
 		_objectActionLocalService.deleteObjectAction(systemObjectAction);
 	}
 
+	@Test
+	public void testExecuteObjectActionMultipleTimesInTheSameThread()
```
[the remark this answers is not in the document]
Comment (devpedrotavares) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1946]: Brian gave this name :grimacing:

[the remark this answers is not in the document]
Comment (linolaoj) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectActionLocalServiceTest.java:1946]: Lets keep it then.

**Comment by devpedrotavares** [comment]
ci:test:sf

**Comment by devpedrotavares** [comment]
ci:test:relevant

**Comment by devpedrotavares** [comment]
ci:test:object

**Pull request closed without merging**